_includes/tutorials/error-handling/kstreams/code/src/main/java/io/confluent/developer/StreamsUncaughtExceptionHandling.java

```diff
@@ -0,0 +1,155 @@
+        envProps.load(input);
+        input.close();
+
+        return envProps;
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        if (args.length < 1) {
+            throw new IllegalArgumentException("This program takes one argument: the path to an environment configuration file.");
+        }
+
+        StreamsUncaughtExceptionHandling tw = new StreamsUncaughtExceptionHandling();
+        Properties envProps = tw.loadEnvProperties(args[0]);
+        Properties streamProps = tw.buildStreamsProperties(envProps);
+
+        // Change this to StreamsConfig.EXACTLY_ONCE to eliminate duplicates
+        streamProps.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, StreamsConfig.AT_LEAST_ONCE);
+        Topology topology = tw.buildTopology(envProps);
+
+        tw.createTopics(envProps);
+        TutorialDataGenerator dataGenerator = new TutorialDataGenerator(envProps);
+        dataGenerator.generate();
+
+        final int maxFailures = Integer.parseInt(envProps.getProperty("max.failures"));
+        final long maxTimeInterval = Long.parseLong(envProps.getProperty("max.time.millis"));
+        final KafkaStreams streams = new KafkaStreams(topology, streamProps);
+        final MaxFailuresUncaughtExceptionHandler exceptionHandler = new MaxFailuresUncaughtExceptionHandler(maxFailures, maxTimeInterval);
+        streams.setUncaughtExceptionHandler(exceptionHandler);
+
+        // Attach shutdown handler to catch Control-C.
+        Runtime.getRuntime().addShutdownHook(new Thread("streams-shutdown-hook") {
+            @Override
+            public void run() {
+                streams.close();
+            }
+        });
+
+        try {
+            streams.cleanUp();
+            streams.start();
+        } catch (Throwable e) {
+            System.exit(1);
+        }
+    }
+
+    static class TutorialDataGenerator {
+        final Properties properties;
+
+
+        public TutorialDataGenerator(final Properties properties) {
+            this.properties = properties;
+        }
+
+        public void generate() {
+            properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+            properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+            try (Producer<String, String> producer = new KafkaProducer<>(properties)) {
+                String topic = properties.getProperty("input.topic.name");
+                List<String> messages = Arrays.asList("All", "streams", "lead", "to", "Confluent", "Go", "to", "Kafka", "Summit");
+
+
+                messages.forEach(message -> producer.send(new ProducerRecord<>(topic, message), (metadata, exception) -> {
+                        if (exception != null) {
+                            exception.printStackTrace(System.out);
+                        } else {
+                            System.out.printf("Produced record at offset %d to topic %s \n", metadata.offset(), metadata.topic());
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.

Reply:
@rubieHess good catch, I've fixed this.

Reply:
Happy to help:)